src/org/parosproxy/paros/core/scanner/Scanner.java

```diff
@@ -256,56 +249,38 @@ public void scan(Target target) {
 		            notifyHostNewScan(pmSet.getKey(), pmSet.getValue());
 		    	}
 		    }
-	    } else if (target.getContext() != null) {
-	    	// Loop through all of the top nodes containing children in this context
-	    	// TODO need to change for lowmem
-	    	if (Constant.isLowMemoryOptionSet()) {
-	    		throw new InvalidParameterException("Not yet supported for the low memory option :(");
-	    	}
-	    	List<SiteNode> nodes = target.getContext().getTopNodesInContextFromSiteTree();
-	    	for (SiteNode node : nodes) {
-			    HostProcess hostProcess = null;
-	            String hostAndPort = getHostAndPort(node);
-	            hostProcess = new HostProcess(hostAndPort, this, scannerParam, 
-	            		connectionParam, scanPolicy, ruleConfigParam);
-	            hostProcess.setStartNode(new StructuralSiteNode(node));
-	            hostProcess.setUser(this.user);
-	            hostProcess.setTechSet(this.techSet);
-	            this.hostProcesses.add(hostProcess);
-	            do { 
-	                thread = pool.getFreeThreadAndRun(hostProcess);
-	                if (thread == null) Util.sleep(500);
-	            } while (thread == null && !isStop());
-	            if (thread != null) {
-		            notifyHostNewScan(hostAndPort, hostProcess);
-	            }
-	    	}
-	    } else if (target.isInScopeOnly()) {
-	    	// TODO need to change for lowmem
-	    	if (Constant.isLowMemoryOptionSet()) {
-	    		throw new InvalidParameterException("Not yet supported for the low memory option :(");
-	    	}
-	    	this.justScanInScope = true;
-	    	List<SiteNode> nodes = Model.getSingleton().getSession().getTopNodesInScopeFromSiteTree();
-	    	for (SiteNode node : nodes) {
-			    HostProcess hostProcess = null;
-	            String hostAndPort = getHostAndPort(node);
-	            hostProcess = new HostProcess(hostAndPort, this, scannerParam, 
-	            		connectionParam, scanPolicy, ruleConfigParam);
-	            hostProcess.setStartNode(new StructuralSiteNode(node));
-	            hostProcess.setUser(this.user);
-	            hostProcess.setTechSet(this.techSet);
-	            this.hostProcesses.add(hostProcess);
-	            do { 
-	                thread = pool.getFreeThreadAndRun(hostProcess);
-	                if (thread == null) Util.sleep(500);
-	            } while (thread == null && !isStop());
-	            if (thread != null) {
-		            notifyHostNewScan(hostAndPort, hostProcess);
-	            }
-	    	}
+	    } else {
+            // TODO need to change for lowmem
+            if (Constant.isLowMemoryOptionSet()) {
+                throw new InvalidParameterException("Not yet supported for the low memory option :(");
+            }
+
+            List<SiteNode> nodes = null;

```

Comment:
Should be initialised to an empty list, otherwise it could lead to a `NullPointerException` when iterating (if neither in scope nor scanning a context).